libtiledbvcf/cmake/Modules/FindHTSlib.cmake

```diff
@@ -95,7 +95,7 @@ if (NOT HTSLIB_FOUND)
       ExternalProject_Add(ep_htslib
         PREFIX "externals"
         URL https://github.com/TileDB-Inc/m2w64-htslib-build/releases/download/1.20-0/m2w64-htslib-1.20-0.tar.gz
-        URL_HASH SHA1=da39a3ee5e6b4b0d3255bfef95601890afd80709
```

Comment:
Why did the mismatch happen? We should investigate it before changing the hash.

Reply:
Unclear exactly how the mismatch was introduced. It was added in https://github.com/TileDB-Inc/TileDB-VCF/pull/746

But it's been known to be a problem for about a month now. I first saw it in https://github.com/TileDB-Inc/TileDB-VCF/pull/763, which I then also used in https://github.com/TileDB-Inc/TileDB-VCF/pull/766

Reply:
This change shouldn't be needed for this PR though. That Windows CI has m2w64-htslib installed from conda. For some reason, in this PR, conda installs m2w64-htslib 1.20, which CMake can't find `-- Could NOT find HTSlib (missing: HTSLIB_LIBRARIES HTSLIB_INCLUDE_DIR)`

Whereas the [last successful Windows CI build](https://github.com/TileDB-Inc/TileDB-VCF/actions/runs/10904790275/job/30262074685) from last week installed m2w64-htslib 1.19 with conda, which CMake was able to find `-- Found HTSlib: C:/Users/runneradmin/micromamba/envs/win-env/Library/lib/hts-3.lib`

So 2 puzzles:

1. Why is conda installing m2w64-1.20 this week but 1.19 just last week?
2. Why can't CMake find htslib 1.20?

Reply:
As a workaround if we can't quickly figure this out, we could pin m2w64-htslib to 1.19

https://github.com/TileDB-Inc/TileDB-VCF/blob/2271db093231a7e3a935da1e5760b7e4cf9c5bc8/ci/gha-win-env.yml#L10

Reply:
The conda binary for 1.20 is smaller compared to 1.19:

![image](https://github.com/user-attachments/assets/8662ddb1-3135-46f6-80fa-523e81da415c)

However, I downloaded the conda binary, and it still has `Library/lib/hts-3.lib`, so I would think that CMake should be able to find it.

Reply:
The existing hash for `m2w64-htslib-1.20-0.tar.gz` was wrong. 

`da39a3ee5e6b4b0d3255bfef95601890afd80709` is the hash from a downloading the wrong URL.

```bash
$ wget -O- https://github.com/TileDB-Inc/m2w64-htslib-build/releases/download/foobar | sha1sum 
--2024-09-27 15:51:28--  https://github.com/TileDB-Inc/m2w64-htslib-build/releases/download/foobar
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-09-27 15:51:29 ERROR 404: Not Found.

da39a3ee5e6b4b0d3255bfef95601890afd80709  -
```